### PR TITLE
Remove utils/bigint in favor of fvm_shared::bigint

### LINKED ISSUE #1554: Remove `utils/bigint` in favor of `fvm_shared::bigint`
**Issue summary**

 - [ ] Update code to use `fvm_shared::bigint` directly.
 - [ ] Remove `utils/bigint` source code.

**Other information and links**

### PR DESCRIPTION
**Pull request by tyshko5**
**Summary of changes**
Changes introduced in this pull request:
- 

**Reference issue to close (if applicable)**

Closes #1554 

**Other information and links**

### CODE REVIEW

File: blockchain/blocks/src/election_proof.rs [hunk 1]
```diff
@@ -4,7 +4,8 @@
 use crypto::VRFProof;
 use encoding::{blake2b_256, tuple::*};
 use fil_types::BLOCKS_PER_EPOCH;
-use num_bigint::{BigInt, Integer, ParseBigIntError, Sign};
+use fvm_shared::bigint::{BigInt, Integer, ParseBigIntError, Sign};
+// use fvm_shared::bigint::{BigInt, Integer, ParseBigIntError, Sign};
```
Comment (lemmih) [inline blockchain/blocks/src/election_proof.rs:8]: You can remove this line.

Comment (LesnyRumcajs) [inline blockchain/blocks/src/election_proof.rs:8]:
```suggestion
```

File: node/rpc-api/src/data_types.rs [hunk 2]
```diff
@@ -24,17 +24,18 @@ use clock::ChainEpoch;
...
     signed_message::json::SignedMessageJson, unsigned_message, SignedMessage, UnsignedMessage,
 };
 use message_pool::{MessagePool, MpoolRpcProvider};
-use num_bigint::{bigint_ser, BigInt};
+use num_bigint::bigint_ser::json;
+use num_bigint::bigint_ser::json::opt;
 use state_manager::{MiningBaseInfo, StateManager};
 use vm::{ActorState, TokenAmount};
 use wallet::KeyStore;
-
```
Comment (lemmih) [inline node/rpc-api/src/data_types.rs:37]: You probably didn't mean to remove the newline between the imports and the code.

File: node/rpc-api/src/data_types.rs [hunk 3]
```diff
@@ -83,7 +84,7 @@ pub struct BlockMessages {
 #[derive(Serialize, Deserialize)]
 #[serde(rename_all = "PascalCase")]
 pub struct MessageSendSpec {
-    #[serde(with = "bigint_ser::json")]
+    #[serde(with = "bigint_ser")]
```
Comment (lemmih) [inline node/rpc-api/src/data_types.rs:87]: This doesn't look right. The json serializer isn't the same as the binary serializer.

File: blockchain/message_pool/src/msgpool/msg_pool.rs [hunk 4]
```diff
@@ -30,12 +30,12 @@ use db::Store;
 use encoding::Cbor;
 use forest_libp2p::{NetworkMessage, Topic, PUBSUB_MSG_STR};
 use futures::{future::select, StreamExt};
+use fvm_shared::bigint::BigInt;
+use fvm_shared::bigint::Integer;
```
Comment (LesnyRumcajs) [inline blockchain/message_pool/src/msgpool/msg_pool.rs:34]:
```suggestion
use fvm_shared::bigint::{BigInt, Integer};
```

File: types/src/genesis/mod.rs [hunk 5]
```diff
@@ -2,8 +2,9 @@
 // SPDX-License-Identifier: Apache-2.0, MIT
 
 use address::{json as addr_json, Address};
+use fvm_shared::bigint::bigint_ser as bigint_json;
```
Comment (lemmih) [inline types/src/genesis/mod.rs:5]: This does not look right. If it _is_ right, please explain why.

File: types/src/genesis/mod.rs [hunk 6]
```diff
@@ -2,8 +2,9 @@
...
 use fvm_shared::sector::SectorSize;
-use num_bigint::bigint_ser::json as bigint_json;
+// use num_bigint::bigint_ser::json as bigint_json;
```
Comment (LesnyRumcajs) [inline types/src/genesis/mod.rs:7]:
```suggestion
```

File: vm/interpreter/src/vm.rs [hunk 7]
```diff
@@ -17,12 +17,14 @@ use forest_encoding::Cbor;
 use fvm::executor::ApplyRet;
 use fvm::machine::NetworkConfig;
 use fvm::machine::{Engine, Machine};
+
+use fvm_shared::bigint::BigInt;
+
```
Comment (LesnyRumcajs) [inline vm/interpreter/src/vm.rs:22]:
```suggestion
use fvm_shared::bigint::BigInt;
```

File: vm/message/src/unsigned_message.rs [hunk 8]
```diff
@@ -253,7 +253,9 @@ pub mod json {
     use super::*;
     use address::json::AddressJson;
     use cid::Cid;
+    // use fvm_shared::bigint::bigint_ser;
```
Comment (lemmih) [inline vm/message/src/unsigned_message.rs:256]: This can be removed.

File: vm/message/src/unsigned_message.rs [hunk 9]
```diff
@@ -253,7 +253,9 @@ pub mod json {
...
     use num_bigint::bigint_ser;
+
```
Comment (lemmih) [inline vm/message/src/unsigned_message.rs:258]: You probably didn't mean to add a newline here.

**Review by lemmih — CHANGES_REQUESTED** [review]

File: vm/interpreter/src/vm.rs [hunk 10]
```diff
@@ -389,8 +391,8 @@ where
             .execute_message(msg.into(), fvm::executor::ApplyKind::Implicit, raw_length)
             .map_err(|e| format!("{:?}", e))?;
         ret.msg_receipt.gas_used = 0;
-        ret.miner_tip = num_bigint::BigInt::zero();
-        ret.penalty = num_bigint::BigInt::zero();
+        ret.miner_tip = fvm_shared::bigint::BigInt::zero();
+        ret.penalty = fvm_shared::bigint::BigInt::zero();
```
Comment (LesnyRumcajs) [inline vm/interpreter/src/vm.rs:395]:
```suggestion
        ret.miner_tip = BigInt::zero();
        ret.penalty = BigInt::zero();
```

File: vm/message/src/unsigned_message.rs (hunk 8 above)
Comment (LesnyRumcajs) [inline vm/message/src/unsigned_message.rs:256]:
```suggestion
```

File: vm/message/src/unsigned_message.rs (hunk 9 above)
Comment (LesnyRumcajs) [inline vm/message/src/unsigned_message.rs:258]:
```suggestion
```

**Review by LesnyRumcajs — CHANGES_REQUESTED** [review]

File: node/rpc-api/src/data_types.rs (hunk 3 above)
Comment (LesnyRumcajs) [reply node/rpc-api/src/data_types.rs:87]: Good catch.

File: types/src/genesis/mod.rs (hunk 5 above)
Comment (tyshko5) [reply types/src/genesis/mod.rs]: Fixed

File: vm/message/src/unsigned_message.rs (hunk 9 above)
Comment (tyshko5) [reply vm/message/src/unsigned_message.rs]: Fixed

File: blockchain/blocks/src/election_proof.rs (hunk 1 above)
Comment (tyshko5) [reply blockchain/blocks/src/election_proof.rs]: Cannot see such a thing

File: vm/interpreter/src/vm.rs (hunk 7 above)
Comment (tyshko5) [reply vm/interpreter/src/vm.rs]: Fixed

File: vm/interpreter/src/vm.rs (hunk 10 above)
Comment (tyshko5) [reply vm/interpreter/src/vm.rs]: Fixed

File: node/rpc-api/src/data_types.rs [hunk 11]
```diff
@@ -186,9 +188,9 @@ pub struct BlockTemplate {
 #[derive(Serialize)]
 #[serde(rename_all = "PascalCase")]
 pub struct MiningBaseInfoJson {
-    #[serde(with = "bigint_ser::json::opt")]
+    #[serde(with = "opt")]
```
Comment (lemmih) [inline node/rpc-api/src/data_types.rs:191]: nit: Maybe rename `opt` to `option` and keep the `json` prefix. Like this: `#[serde(with = "json::option")]`

Comment (tyshko5) [reply node/rpc-api/src/data_types.rs]: Fix

**Review by LesnyRumcajs — APPROVED** [review]
nice!

**Pull request merged**